src/engine/dag.rs

```diff
@@ -248,23 +248,24 @@ impl Dag {
     }
 
     /// This function is used for the execution of a single dag.
-    pub fn start(&mut self) -> Result<bool, DagError> {
+    pub fn start(&mut self) -> Result<(), DagError> {
         // If the current continuable state is false, the task will start failing.
         if self.can_continue.load(Ordering::Acquire) {
             self.init().map_or_else(Err, |_| {
-                Ok(tokio::runtime::Runtime::new()
+                tokio::runtime::Runtime::new()
                     .unwrap()
-                    .block_on(async { self.run().await }))
+                    .block_on(async { self.run().await })
             })
         } else {
-            Ok(false)
+            // TODO: Change this error
+            Err(DagError::EmptyJob)
```

Comment:
What error would be fitting here? We could add another enum variant, but I'm not sure about the right name.